# Key file types in genomic epidemiology workflow with Illumina sequencing 

Genomic epidemiology relies on knowing the mutations on individual genomes 

| data analysis | step 1 | step 2 | step 3 | step 4 | 
|---|-----|------|-------|--|
| file content | raw sequencing reads | reads mapped to reference genome | consensus genome | tree |
| file type | fastq file | bam file | fasta file | json file |

### Read file: `sample.fastq` or `sample.fq`

- The file generated by the sequencer is usually in FASTQ format. FASTQ files contains the raw sequencing reads and their quality scores.
- Below is 1 read in the FASTQ file. 
```
@A00111:640:HVVF3DRXX:1:2101:30038:1063 1:N:0:CTCTGATTGCCT+TTGGTCTGCCGA
CTATCATTAACTGAATCCATAGGTTAATGAGGCGAACCGGGGGAACTGAAACATCTAAGTACCCCGAGGAAAAGAAATCAACCGAGATTCCCCCAGTAGCGGCGAGCGAACGGGGAGCAGCCCAGAGCCTGAATCAGTGTGTGTGC
+
FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFF:FFF:FFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:F
```
-  
   + Row 1: read name starting with @
   + Row 2: actual nucleotide sequence of the read
   + Row 3: usually a + sign
   + Row 4: quality (confidence) of each base. It has to be the same length as row 2.

So each read is represented by 4 rows. If one FASTQ file contains 1000 reads, it will have 4 x 1000 = 4000 rows.

- If the sequencing is done in single-end mode, there will be 1 FASTQ file for each library; if it is done in paired-end mode, there will be 2 FASTQ files, one for each read in the read pair, usually named as `sample_R1.fastq` and `sample_R2.fastq`.

- A real example from SRA (the NIH repository for sequencing read data): 

https://sra-pub-sars-cov2.s3.amazonaws.com/sra-src/SRR13690103/RR065e_01847_W-R175-A1_B01_covid_1.fq.gz.1

https://sra-pub-sars-cov2.s3.amazonaws.com/sra-src/SRR13690103/RR065e_01847_W-R175-A1_B01_covid_2.fq.gz.1  

- Further reading: https://support.illumina.com/bulletins/2016/04/fastq-files-explained.html


### Mapped read file: `sample.bam`

- To characterize an individual genome, aka to identify the mutations, we need to "compare" the sequencing reads originated from this genome to the reference genome.

- Such "comparison" results in a Bam file that contains comprehensive information of how each sequencing read compares with the reference genome. 

- Bam file is binary to reduce file size, so requires special tools to open and read. 

- This is an essential step but bam files are usually not directly used in genomic epidemiology workflow.


### Consensus genome: `sample.fasta` or `sample.fa`

- With the information in a bam file, one can re-constitute the full genome that was sequenced. 

- Genomic sequences, constituting of A, T, C, G or [degenerated nucleotides](https://genome.ucsc.edu/goldenPath/help/iupac.html), are stored in FASTA files.

- A FASTA file can contain 1 or many genomes.

- Below is an example FASTA file with 2 genomes:

```
>sample1_name
NNNNNNNNNCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCACTC
GGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAACNNNNN

>sample2_name
NNNNNNNNNCTCTTGTAGATCTGTTCTCTAAACGAACTTTAAAATCTGTGTGGCTGTCAC
TCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATANNNNN
```
-
  + Rows starting with `>` contain the name of the genomes proceeding.
  + The beginning and end of genomes are sometimes not covered in sequencing due to technical reasons, or have low read quality, therefore are filled with N.

- This is the file people submit to GISAID or NCBI GenBank.


### Phylogeny tree: 

